engine/apps/slack/tasks.py

```diff
@@ -142,15 +143,27 @@ def check_slack_message_exists_before_post_message_to_thread(
             ).save()
 
 
-@shared_dedicated_queue_retry_task(autoretry_for=(Exception,), retry_backoff=True, max_retries=1)
-def send_message_to_thread_if_bot_not_in_channel(alert_group_pk, slack_team_identity_pk, channel_id):
+@shared_dedicated_queue_retry_task(
+    autoretry_for=(Exception,),
+    dont_autoretry_for=(Retry,),
```

Comment:
we already follow a similar pattern of [`dont_autoretry_for`](https://github.com/grafana/oncall/blob/b19cdc8b0c7df0e080165a2e2f3fd5d001c49814/engine/apps/alerts/tasks/notify_user.py#L241) + [`raise <task_name>.retry`](https://github.com/grafana/oncall/blob/b19cdc8b0c7df0e080165a2e2f3fd5d001c49814/engine/apps/alerts/tasks/notify_user.py#L301)